### ISSUE TITLE
Custom tabindex in modal does not work

### ISSUE THREAD

**Issue by Voodu** [bootstrap-vue/bootstrap-vue#6413]
### Describe the bug

Elements with `tabindex` > 0 in the modal are never activated.

### Steps to reproduce the bug

1. Create a modal with elements having `tabindex` > 0, ex:
```html
<template>
    <b-modal title="Some title" id="edit-text-element-modal">
        <input tabindex=0 type="text" placeholder="Some input which can be tabbed">
        <template v-slot:modal-footer>
            <button tabindex="2">
                Cancel
            </button>
            <button tabindex="1">
                OK
            </button>
        </template>
    </b-modal>
</template>
```
2. Display it somehow, ex.
```javascript
    mounted() {
        this.$bvModal.show("edit-text-element-modal");
    }
```
3. When modal is opened, try to navigate with TAB through all the tabable elements (here: `input` and two `button`s)
4. Elements with `tabindex` > 0 are never focused.
### Expected behavior

Elements with `tabindex` > 0 are focused in the correct order, i.e. "OK" button, "Cancel" button, input field.

### Versions

**Libraries:**

- BootstrapVue: 2.14.0
- Vue: 2.6.11

**Environment:**

- Device: Lenovo Thinkpad
- OS: Ubuntu 20.04.2 LTS
- Browser: Firefox
- Version: 85.0 x64

### Additional context

In short, this behavior caused problems when I wanted to change tab order of the buttons (make OK button get focus first even if it's later in the DOM).

After looking through the https://github.com/bootstrap-vue/bootstrap-vue/blob/dev/src/components/modal/modal.js, I think it's caused by tabTraps with `tabindex="0"`. TAB index starts at index 0 and cannot leave that "0 level", because when it is about to do so, it's reset to one of the tabTraps at stays at 0.

**Comment by Hiws** [comment]
https://bootstrap-vue.org/docs/components/modal#keyboard-navigation
> Avoid setting tabindex on elements within the modal to any value other than 0 or -1. Doing so will make it difficult for people who rely on assistive technology to navigate and operate page content and can make some of your elements unreachable via keyboard navigation.

The docs does mention that there can be potential issues if you use a tabindex other than `-1` and `0`, and make certain element unreachable . Which is what I'm guessing is happening here.

Whether this is intended or a just a bug that wasn't fixed I can't say

**Comment by andrei-gheorghiu** [comment]
Here's how you can achieve it: https://jsfiddle.net/websiter/mfL167jd/

It's setting `tabindex` to `-1` for all elements you want to exclude, setting footer buttons `tabindex=0`. Note their order in DOM should be the order in which you want the tabindex to select them and using CSS flexbox `order` to change their display order. It should work cross browser (but I haven't tested it outside Chrome).

I woudn't recommend setting `tabindex` values above `0` as in production scenarios you're likely to have other elements with `tabindex=0` in page and you might get weird results tabbing through elements in page while modal is open.

**Comment by Hiws** [comment]
> Here's how you can achieve it: https://jsfiddle.net/websiter/mfL167jd/
> 
> It's setting `tabindex` to `-1` for all elements you want to exclude, setting footer buttons `tabindex=0`. Note their order in DOM should be the order in which you want the tabindex to select them and I'm also using CSS flexbox `order` to change their display order. It should work cross browser (but I haven't tested it outside Chrome).
> 
> I woudn't recommend setting `tabindex` values above `0` as in production scenarios you're likely to have other elements with `tabindex=0` in page and you might get weird results tabbing through elements in page while modal is open.

I can't seem to tab to the `close` button or `input` in your fiddle, is that intended?

You could also add the classes `flex-row-reverse justify-content-start` to the `footer-class` prop, and swap the position on the DOM to receive the effect of the OK button being first, and then the cancel button.

<https://codepen.io/Hiws/pen/WNorERd>

**Comment by andrei-gheorghiu** [comment]
> I can't seem to tab to the `close` button or `input` in your fiddle, is that intended?

Yep, it's intended. I've set their tabidex values to `-1` (a.k.a: skipped by tabbing - but programmatically focusable). If I leave them at `0` they're focused before the OK button, which seems to be what OP wants. I haven't tried setting them to  `1`, though.

**Comment by Voodu** [comment]
> Here's how you can achieve it: https://jsfiddle.net/websiter/mfL167jd/
> 
> It's setting `tabindex` to `-1` for all elements you want to exclude, setting footer buttons `tabindex=0`. Note their order in DOM should be the order in which you want the tabindex to select them and I'm also using CSS flexbox `order` to change their display order. It should work cross browser (but I haven't tested it outside Chrome).
> 
> I woudn't recommend setting `tabindex` values above `0` as in production scenarios you're likely to have other elements with `tabindex=0` in page and you might get weird results tabbing through elements in page while modal is open.

That CSS `order` should solve my problem. Thank you.
